### ISSUE TITLE
Faster source mirror

### ISSUE THREAD

**Issue by haampie** [LLNL/spack#23062]
The spack default source mirror provides very low download speeds (at least in Europe). I have a 1Gb/s internet connection at home and usually reach around peak speeds, yet "fetch" times are a large part of installing a package:

```
zlib:          Fetch: 2.37s.      Build: 0.87s.      Total: 3.23s.
libfabric:     Fetch: 6.23s.      Build: 9.54s.      Total: 15.76s.
berkeley-db:   Fetch: 12.91s.     Build: 17.70s.     Total: 30.61s.
cpio:          Fetch: 4.73s.      Build: 24.07s.     Total: 28.80s.
cuda:          Fetch: 4m 27.95s.  Build: 1m 22.58s.  Total: 5m 50.52s.
```

Can someone look into this?

**Comment by haampie** [comment]
When I disable the source mirror I get the following fetch times (6x or 7x faster):

```
libfabric: 0.84s
cuda: 42.27s
```

**Comment by vvolkl** [comment]
Can confirm that the source mirror is (sometimes? just recently?) quite slow in europe.

**Comment by tgamblin** [comment]
Fixed by #23978.
